### PR TITLE
Fix the single qubit op wires for CNOT & handle requests with ERROR

### PR DESCRIPTION
**Pull request by antalszava**
**Context**

A minor error in the way how wires are defined for single-qubit operations caused a request error when using the CNOT operation.

**Changes**

* Adjusted the wires in the decomposition of CNOT
* Added a case for handling requests that return a message with the `ERROR` field specified

**Related issues**

#17

### CODE REVIEW

**Comment by antalszava** [comment]
[ch8935]

File: pennylane_aqt/device.py [hunk 1]
```diff
@@ -178,6 +178,13 @@ def apply(self, operations, **kwargs):
             job = submit(self.HTTP_METHOD, self.hostname, job_query_data, self.header).json()
             sleep(self.retry_delay)
 
+        error_msg = job.get("ERROR", None)
+
+        if error_msg:
```
Comment (rmoyard) [inline pennylane_aqt/device.py:183]: Nice to have, it is useful for debugging!

File: tests/test_device.py [hunk 2]
```diff
@@ -578,3 +598,15 @@ def test_simulator_default_init(self, num_wires, shots):
         assert dev.HTTP_METHOD == HTTP_METHOD
         assert API_HEADER_KEY in dev.header.keys()
         assert dev.header[API_HEADER_KEY] == SOME_API_KEY
+
+    @pytest.mark.skip("API key needs to be inputted")
+    def test_simulator_default_init(self, num_wires, shots):
```
Comment (rmoyard) [inline tests/test_device.py:603]:
```suggestion
    def test_simulator_cnot(self, num_wires, shots):
```

**Review by rmoyard — COMMENTED** [review]
It seems good to me 💯 As you mentioned, it would be great to have more integration tests!

**Comment by antalszava** [comment]
Hi @rmoyard, thanks for the review! Indeed, more integration tests would be great. However, I think that will be a different type of push to this specific change. Maybe it's best to have this in and add tests that submit jobs to the remote backend separately.

**Review by rmoyard — APPROVED** [review]

**Pull request merged**